node/src/actors/chain_manager/mining.rs

```diff
@@ -314,6 +332,53 @@ impl ChainManager {
                     }
                 })
                 .flatten()
+                // Refrain from trying to resolve any more requests if we have already hit the limit
+                // of retrievals per epoch.
+                .and_then(move |vrf_proof| {
+                    let mut start_retrieval_count = cloned_retrieval_count.load(Relaxed);
+                    let mut final_retrieval_count = start_retrieval_count.saturating_add(added_retrieval_count);
+
+                    if final_retrieval_count > maximum_retrieval_count {
```

Comment:
Ok, so if `maximum_retrieval_count` is `u16::MAX`, then this branch is unreachable and there is literally no limit. If that's the expected behavior add a short comment somewhere.